crates/bevy_text/src/text.rs

```diff
@@ -351,11 +354,76 @@ impl Default for TextFont {
             font: Default::default(),
             font_size: 20.0,
             line_height: LineHeight::default(),
+            font_features: FontFeatures::default(),
             font_smoothing: Default::default(),
         }
     }
 }
 
+/// OpenType features for .otf fonts that support them.
+///
+/// Examples features include:
+/// "liga": Standard ligatures
+/// "clig": Contextual ligatures
+/// "dlig": Discretionary ligatures
+/// "smcp": Small-caps
+/// "ss01", "ss02", ..., "ss20": Stylistic alternates
+/// "frac": Fractions, formatting numbers like 1/2
+/// "ordn": Ordinals, formatting characters like "1st" or "2nd" properly
+/// For the complete list of OpenType features, see the spec at
+/// `<https://learn.microsoft.com/en-us/typography/opentype/spec/featurelist>`.
+///
+/// The above font features can be enabled via [`FontFeatures::enable`]. Some font features take
+/// numeric values, rather than being on/off switches. For example, for weights of variable
+/// fonts, the font feature "wght" can be any value between 100 and 900, with lower values
+/// corresponding to lighter weights and higher values corresponding to heavier weights. For
+/// these features, use [`FontFeatures::set`] to assign a specific value.
+#[derive(Clone, Debug, Default, Reflect)]
+pub struct FontFeatures {
+    features: HashMap<[u8; 4], u32>,
```

Comment:
I'm not sure about using a hashmap, maybe we should just use a vec like they do in cosmic text and rustybuzz. Maybe it doesn't matter though.

Reply:
Sure, I can switch this to a vec. My original thinking is that something like this would probably result in undefined behavior:

```
FontFeatures::new()
  .set(b"wght", 200)
  .set(b"wght", 600)
```

With a HashMap, we'd always respect the most recently set value. But it might be overkill to try to prevent that here, and instead should just defer to cosmic-text to decide what should happen.

Reply:
It's a list that's applied in order, with duplicates overwriting the previous value so there's no problem, except the obvious worry that someone naively uses `set` or `enable` every frame leaking a huge list of features.

Maybe we could remove the `enable` and `set` functions and instead change it so that `FontFeatures` has to be constructed from a builder, iterator or `From` impl with the list of features being immutable after construction? Font features are almost always just set once at creation, with in-place mutation being relatively rare.

Reply:
Thanks @ickshonpe! That makes sense. I just updated the PR, so now `FontFeatures` must either be constructed from `FontFeaturesBuilder` or an `IntoIterator` `From` impl.

Reply:
Maybe the `FontFeaturesBuilder` constructor should be on `FontFeatures` for better discoverability, so instead of:
```
font_features: FontFeaturesBuilder::new().enable(SLASHED_ZERO).build()
```
you'd use:
```
font_features: FontFeatures::new().enable(SLASHED_ZERO).build()
```

Sorry I'm being way too fussy over this. Perhaps the hashmap was the right approach. If rust supported varidics this would be trivial hehe.